File: prototypes/angular-node-javascript/backend-node-javascript/src/features/person/person-service.js
class PersonService {
  constructor(repository) {
    this.repository = repository;
  }

  async getItems() {
    return await this.repository.getItems();
  }

  async getItemById(id) {
    return await this.repository.getItemById(id);
  }

  async createItem(person) {
    return await this.repository.createItem(person);
  }

  async updateItem(id, updatedData) {
    return await this.repository.updateItem(id, updatedData);
  }

  async deleteItem(id) {
    return await this.repository.deleteItem(id);
  }
}

export default PersonService;

File: backend-java/03-backend-java/src/main/java/com/ganatan/modules/person/PersonService.java

package com.ganatan.modules.person;

import java.util.List;

public class PersonService {

    private final PersonRepository repository;

    public PersonService(PersonRepository repository) {
        this.repository = repository;
    }

    public List<Person> getItems() {
        return repository.getItems();
    }
}
